### ISSUE TITLE
Error with imenu

### ISSUE THREAD

**Issue by gjm** [ejmr/php-mode#404]
When loading php-mode i get this error:

`Error in menu-bar-update-hook (imenu-update-menubar): (void-function semantic-create-imenu-index)
imenu--make-index-alist: Symbol’s function definition is void: semantic-create-imenu-index`

**Comment by zonuexe** [comment]
@gjm Thanks for reporting!
Probably this problem has been fixed in `master`.

**Comment by gjm** [comment]
Not completely fixed... Now i get:

`imenu-unavailable-error: imenu unavailable: "No items suitable for an index found in this buffer"`

**Comment by fallchildren** [comment]
@gjm Quick fix would be to set `php-do-not-use-semantic-imenu` to t.

@zonuexe 
This was introduced because of my merge pull-request #400. Which is a bit more tricky than I thought. 

The variable `imenu-create-index-function` is set mode-locally to `semantic-create-imenu-index` in "semantic/bovine/c". Unless this file is loaded the value is `imenu-default-create-index-function`, which works for stock php-mode. Maybe we can do two things to make this okay for most of the users:

- Set default value of `php-do-not-use-semantic-imenu`  to t (which most users woul want anyway, i guess)
- Require semantic/imenu only if  `php-do-not-use-semantic-imenu` is set to nil

I will make a pull request to address this. Sorry for the bug I did not notice this one as I tested only with semantic-mode enabled.
